tools/cli/internal/changelog/outputfilter/squash.go

```diff
@@ -0,0 +1,211 @@
+package outputfilter
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"regexp"
+)
+
+var (
+	// identifierRegex matches text enclosed in single quotes.
+	// Example: it'll match "value" from the string "added the new required request property 'value'".
+	identifierRegex = regexp.MustCompile(`'([^']+)'`)
+)
+
+type SquashHandler interface {
+	Squash(entriesGroupedByOperationID map[string][]*Entry) []*Entry
+}
+
+func newSquashHandlers() map[string]func(map[string][]*Entry) ([]*Entry, error) {
+	return map[string]func(map[string][]*Entry) ([]*Entry, error){
+		// field changes
+		"response-required-property-added":            SquashResponseRequiredFieldAdded,
+		"response-required-property-removed":          SquashResponseRequiredFieldRemoved,
+		"response-optional-property-added":            SquashResponseOptionalFieldAdded,
+		"response-optional-property-removed":          SquashResponseOptionalFieldRemoved,
+		"response-property-became-required":           SquashResponseFieldBecameRequired,
+		"request-property-became-required":            SquashRequestFieldBecameRequired,
+		"new-required-request-property":               SquashRequestFieldAdded,
+		"request-property-removed":                    SquashRequestFieldRemoved,
+		"new-optional-request-property":               SquashNewOptionalRequestProperty,
+		"response-optional-property-became-read-only": SquashResponseOptionalFieldBecomeReadonly,
+	}
+}
+
+// EntryMappings groups entries by ID and then by OperationID
+func newEntriesMapPerIDAndOperationID(entries []*Entry) map[string]map[string][]*Entry {
```

Comment:
Since this struct is a bit counter-intuitve, can we add:
Map[changeCode, Map[operationId, List[oasdiffEntry]]]

Reply:
I am not sure I understand where I should add it, do you mean as a comment?

Reply:
I added a comment